### DISCUSSION TITLE
Need help: Unable to choose platform folder (Preferences > Java)

### DISCUSSION THREAD

**Discussion by aloraine** [apache/netbeans#4224] [Q&A] [answered]
Hello,
I have just downloaded Netbeans 14 and am trying to use it with an existing maven project for an application that requires JDK 1.8 and JavaFX to run.
For my first step, I tried to configure Netbeans 14 to use a different-from-default JDK/JRE. Specifically, I want it to use a JDK/JRE from Zulu that includes JavaFx, which I have already installed on my system (MacOS). I am able to use this JDK/JRE combination to build and run my application using my homebrew-installed maven software, installed separately from NetBeans 14. 
To try to add this JDK/JRE to my Netbeans 14, I selected NetBeans > Preferences > Maven and clicked "Manage Java Platforms" and then clicked "Add Platform..."
This opened the "Add Java Platform" wizard. I selected the option "Java Standard Edition" and then "Next". I then selected the folder corresponding to the "JAVA_HOME" for the JDK/JRE I want to use.
 However, no matter what I do, the "Next >" button fails to become clickable! For example, I tried selecting sub-folders within my this directory, but nothing worked. Nothing seems to enable the "Next >" button to become clickable. 
I've tried various searches to see if I can find a solution to this problem, but have not come up with anything thus far. So I'm posting my question as a new Discussion in hopes of finding a solution from the NetBeans community.
Thank you so much for this software! I'm really looking forward to using NetBeans 14 for my projects!
-Ann L.

**Comment by aloraine** [comment, marked as answer]
I'm leaving the question here in case it's useful to others! It turns out the reason for my problem was that I had installed only the JRE and not the JDK. I did not realize that JDK includes all the API classes and code *plus* JRE (and compiler, etc) whereas JRE contains only the JVM. When the NetBeans interface fails to recognize a folder as a platform folder, that's because it's not actually a platform folder. At least, that's where I went wrong in this particular case!

**Reply by mbien** [reply]
as developer, the JDK is always what you want to install (or simply extract if you don't want to install it) - as you mentioned, it includes everything.

JREs are less common now since java modules also came with tools like jpackage, jdeps or jlink [1] which allow applications to make their own "runtime". It is also not clearly defined what a general purpose JRE actually is. What if your application would like to use the compiler API? This would not be in a traditional JRE. JREs are basically the legacy of applets, webstart (and bluray players) ;)

I believe its ok for NetBeans to require a JDK since the users are essentially devs, but lets say you write the next minecraft, please don't make everyone install JREs again :)

[1] can be even used without having modules in your application if you are willing to go some extra steps ([blog entry](https://mbien.dev/blog/entry/custom-java-runtimes-with-jlink))
